compaction/compaction_manager.hh

```diff
@@ -75,8 +91,15 @@ private:
         bool compaction_running = false;
         utils::UUID output_run_identifier;
         sstables::compaction_data compaction_data;
+        compaction_state& compaction_state;
+        gate::holder gate_holder;
 
-        explicit task(column_family* cf, sstables::compaction_type type) : compacting_cf(cf), type(type) {}
+        explicit task(column_family* cf, sstables::compaction_type type, struct compaction_state& cs)
```

Comment:
we can drop the keyword struct here

Reply:
no, since I called the member `compaction_state` so `struct` is required to disambiguate the type name from the member name.